libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2953,7 +2953,8 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 
 string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayType const& _to)
 {
-	solUnimplementedAssert(_to.location() != DataLocation::CallData, "Conversion of calldata types not yet implemented.");
+	solAssert(_to.location() != DataLocation::CallData, "");
```

Comment:
As we spoke, there shouldn't be other conversions to calldata in case of arrays except when types are exaclt the same, which ends up in not calling this function at all. So this should be `solAssert`

Reply:
Ah right!